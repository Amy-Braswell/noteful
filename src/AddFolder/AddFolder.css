.AddFolder {
    margin: 130px 0 0;
    padding: 0;
    float: left;
    text-align: left;
    width: 40%;
  }

.AddFolder h3 {
    padding: 0 15px;
    color: #bdbdbd;
}

.error__message {
  padding: 0 15px;
  font-weight: 600;
  color: #f6142f;
  display: block
}

.form-group input{
    background-color: #030f1f;
    border: none;
    color: #b5b3b3; 
    display: block;
    font-family: var(--custom-font);
    font-size: 20px;
    margin: 0 0 5vh;
    width: 100%;
    /* max-width: 180px; */
    padding: 8px 16px;
}


input:-webkit-autofill::first-line {
    font-size: 20px
}

input:-webkit-autofill, input:-webkit-autofill:hover, input:-webkit-autofill:focus,
input, input[type=text] {
    border: none;
    font-size: 20px;
    font-family: var(--custom-font);
    padding: 8px 16px;
    background-color: #030f1f;
    -webkit-box-shadow: 0 0 0px 1000px  #030f1f inset;
    -webkit-text-fill-color: #b5b3b3;
    color: #b5b3b3;
    box-shadow: 0 0 0px 1000px  #030f1f inset;
    outline: none;
}

.buttons button {
    background-color: #cacaca;
    border-radius: 30px;
    border: 0;
    color: black;
    display: block;
    font-size: 12px;
    font-weight: bold;
    height: 60px;
    padding: 8px;
    text-align: center;
    text-decoration: none;
    width: 60px;
    margin: 1vh auto
  }
  
  a.buttons button {
    padding: 16px 8px 8px;
  }
  
  .buttons button:hover {
    background-color: #29862e;
    cursor: pointer;
  }

  .buttons button:focus {
    background-color: #29862e;
  }
  