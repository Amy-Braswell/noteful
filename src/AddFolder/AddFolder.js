import React, { Component } from 'react'
import NotefulForm from '../NotefulForm/NotefulForm'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'
import './AddFolder.css'
import ApiContext from '../ApiContext'

export default class AddFolder extends Component {
  constructor() {
    super()
    this.state = {
      error: null,
      name: '',
      nameValid: false,
      validationMessage: ''
    };
  }
  static contextType = ApiContext

  isNameValid = (e) => {
    e.preventDefault()
    if (!this.state.name) {
      this.setState({
        validationMessage: 'Please enter a folder name',
        nameValid: false
      })
    } 
    else if (this.state.name.length < 3) {
      this.setState({
        validationMessage: 'Folder name must be at least 3 characters long',
        nameValid: false
      })
    }
    else if (this.state.name.length > 10) {
      this.setState({
        validationMessage: 'Folder name cannot be longer than 10 characters',
        nameValid: false
      })
    }
    
    else {
      this.setState(
        {
          validationMessage: '',
          nameValid: true
        },
        this.handleAddFolder()
      )
    }
  }

  handleAddFolder = () => {
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        name: this.state.name
      })
    }

    fetch('http://localhost:8000/api/folders', options)
      .then(res => {
        if (!res.ok) {
          throw new Error('Something went wrong')
        }
        return res
      })
      .then(res => res.json())
      .then(data => {
        this.context.handleAddFolder(data)
      })
      .catch(err => {
        this.setState({
          error: err.message
        })
      })
  }

  updateName = (name) => {
    this.setState({ name: name })
  }

  render() {
    return (
      <section className='AddFolder'>
        <h3>Create a folder</h3>

        {!this.state.nameValid && (
          <div>
            <p className='error__message'>{this.state.validationMessage}</p>
          </div>
        )}

        <NotefulForm
          onSubmit={event => {
            this.isNameValid(event)
          }}
        >
        <div className='form AddFolder_form-group'>
          <div className='field form-group'>
            <label htmlFor="folder-name">
              <input
                type='text'
                className='new__folder__name' 
                name='folder'
                aria-required="true"
                onChange={event => this.updateName(event.target.value)}
              />
            </label>
          </div>
          <div className='buttons'>
            <button type='submit'>
            <FontAwesomeIcon icon='plus' />
              <br />
              Folder
            </button>
          </div>
        </div>
        </NotefulForm>
        {this.state.error && (
          <div>
            <p className='error__message'>{this.state.error}</p>
          </div>
        )}
      </section>
    )
  }
}

AddFolder.propTypes = {
  history: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
}
