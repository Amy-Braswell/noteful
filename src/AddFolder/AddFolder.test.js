import React from 'react';
import renderer from 'react-test-renderer'
import AddFolder from './AddFolder'
import ReactDOM from 'react-dom'

//Smoke Test
it('renders without crashing', () => {
    // start by creating an element to render the component into
    const div = document.createElement('div')
    ReactDOM.render(<AddFolder />, div)
    // Render the component -- Line 9 is the actual test
    // If something is wrong it will fail here.
    ReactDOM.unmountComponentAtNode(div)
    });

describe(`AddItemForm component`, () => {
  it('renders the complete form', () => {
    const wrapper = renderer
    .create(<AddFolder />)
    .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})