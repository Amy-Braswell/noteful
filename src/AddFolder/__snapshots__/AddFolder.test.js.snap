// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm component renders the complete form 1`] = `
<section
  className="AddFolder"
>
  <h3>
    Create a folder
  </h3>
  <div>
    <p
      className="error__message"
    >
      
    </p>
  </div>
  <form
    action="#"
    className="Noteful-form "
    onSubmit={[Function]}
  >
    <div
      className="form AddFolder_form-group"
    >
      <div
        className="field form-group"
      >
        <label
          htmlFor="folder-name"
        >
          <input
            aria-required="true"
            className="new__folder__name"
            name="folder"
            onChange={[Function]}
            type="text"
          />
        </label>
      </div>
      <div
        className="buttons"
      >
        <button
          type="submit"
        >
          <br />
          Folder
        </button>
      </div>
    </div>
  </form>
</section>
`;
